browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -2036,24 +2039,27 @@ void RewardsDOMHandler::OnGetWalletPassphrase(const std::string& passphrase) {
 }
 
 void RewardsDOMHandler::GetOnboardingStatus(const base::Value::List& args) {
-  if (!rewards_service_) {
-    return;
-  }
   AllowJavascript();
+  Profile* profile = Profile::FromWebUI(web_ui());
   base::Value::Dict data;
-  data.Set("showOnboarding", rewards_service_->ShouldShowOnboarding());
+  data.Set("showOnboarding",
+           !profile->GetPrefs()->GetBoolean(brave_rewards::prefs::kEnabled));
   CallJavascriptFunction("brave_rewards.onboardingStatus",
                          base::Value(std::move(data)));
 }
 
-void RewardsDOMHandler::SaveOnboardingResult(const base::Value::List& args) {
-  CHECK_EQ(1U, args.size());
-  if (!rewards_service_)
-    return;
-
+void RewardsDOMHandler::EnableRewards(const base::Value::List& args) {
   AllowJavascript();
```

Comment:
Totally minor, but should we just move this into the `!BUILDFLAG(IS_ANDROID)` section?

Reply:
Done.